applications/GeoMechanicsApplication/custom_elements/line_interface_element.cpp

```diff
@@ -0,0 +1,277 @@
+}
+
+void LineInterfaceElement::CalculateOnIntegrationPoints(const Variable<Vector>& rVariable,
+                                                        std::vector<Vector>&    rOutput,
+                                                        const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rVariable == STRAIN) {
+        const auto local_b_matrices = CalculateLocalBMatricesAtIntegrationPoints();
+        rOutput = CalculateRelativeDisplacementsAtIntegrationPoints(local_b_matrices);
+    } else if (rVariable == CAUCHY_STRESS_VECTOR) {
+        const auto local_b_matrices = CalculateLocalBMatricesAtIntegrationPoints();
+        const auto relative_displacements = CalculateRelativeDisplacementsAtIntegrationPoints(local_b_matrices);
+        rOutput = CalculateTractionsAtIntegrationPoints(relative_displacements);
+    } else {
+        BaseType::CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+        BaseType::CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+    }
+}
+
+void LineInterfaceElement::CalculateOnIntegrationPoints(const Variable<ConstitutiveLaw::Pointer>& rVariable,
+                                                        std::vector<ConstitutiveLaw::Pointer>& rOutput,
+                                                        const ProcessInfo&)
+{
+    KRATOS_ERROR_IF_NOT(rVariable == CONSTITUTIVE_LAW)
+        << "Cannot calculate on integration points: got unexpected variable " << rVariable.Name() << "\n";
+
+    rOutput = mConstitutiveLaws;
+}
+
+Element::Pointer LineInterfaceElement::Create(IndexType               NewId,
+                                              const NodesArrayType&   rNodes,
+                                              PropertiesType::Pointer pProperties) const
+{
+    return Create(NewId, this->GetGeometry().Create(rNodes), pProperties);
+}
+
+Element::Pointer LineInterfaceElement::Create(IndexType               NewId,
+                                              GeometryType::Pointer   pGeometry,
+                                              PropertiesType::Pointer pProperties) const
+{
+    return {make_intrusive<LineInterfaceElement>(NewId, pGeometry, pProperties)};
+}
+
+void LineInterfaceElement::GetDofList(DofsVectorType& rElementalDofList, const ProcessInfo&) const
+{
+    rElementalDofList = GetDofs();
+}
+
+void LineInterfaceElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    BaseType::Initialize(rCurrentProcessInfo);
+
+    const auto unused_shape_function_values = Vector{};
+    for (auto i = std::size_t{0}; i < mIntegrationScheme->GetNumberOfIntegrationPoints(); ++i) {
+        mConstitutiveLaws.push_back(GetProperties()[CONSTITUTIVE_LAW]->Clone());
+        mConstitutiveLaws.back()->InitializeMaterial(GetProperties(), GetGeometry(), unused_shape_function_values);
+    }
+}
+
+Element::DofsVectorType LineInterfaceElement::GetDofs() const
+{
+    const auto no_Pw_geometry = Geometry<Node>{};
```

Comment:
Perhaps even clearer:
```suggestion
    const auto no_Pw_geometry_yet = Geometry<Node>{};
```